monitor/data_monitoring.py

```diff
@@ -156,23 +151,41 @@ def generate_report(self, days_back: Optional[int] = None, test_runs_amount: Opt
                 elementary_output_json_file:
             elementary_output_json_file.write(dumped_output_data)
 
-        webbrowser.open_new_tab('file://' + html_path)
+        if should_open_browser:
+            webbrowser.open_new_tab('file://' + html_path)
         self.execution_properties['report_end'] = True
         self.execution_properties['success'] = self.success
         return self.success, html_path
 
-    def send_report(self, elementary_html_path: str) -> bool:
-        if os.path.exists(elementary_html_path):
+    def send_report(self, html_path: str) -> bool:
+        if self.slack_client:
+            self.execution_properties['send_to_slack'] = True
```

Comment:
let's add at the end of the if statement `self.execution_properties['sent_to_slack_successfully'] = file_uploaded_successfully`, just like you did with uploading to s3 and gcs 🙂

Reply:
There's not much point because we'd see the command is `send-report` and we have the `success` variable which is equivalent.

Reply:
If we'll join the methods it will 🙂

Plus adding this info to the execution_properties can't hurt, and will make it more agnostic to changes in the future (it is likely to forget adding this data due to a change in the way we call the method in the future + if send-report cli command will have more steps in the future that take place after sending the report and they will fail, it will look like we failed to send the report although we did send it).

Plus Plus it is nice that all the information you need is under the same "summary" object. That way you don't have to think "Oh I ran send-report so I need to look at the final status".